### ISSUE TITLE
[BUG]: EDG does not link to libfmt

### ISSUE THREAD

**Issue by dangelog** [compiler-explorer/compiler-explorer#6343]
### Describe the bug

When using libfmt on CE with the EDG frontend, the program compiles, but the linker complains about the library not being found:

```
Could not execute the program
Compiler returned: 2
Compiler stderr

library "fmtd" does not exist in the specified library directories
C++ prelinker: error: command line error
/opt/compiler-explorer/gcc-13.2.0/bin/../lib/gcc/x86_64-linux-gnu/13.2.0/../../../../x86_64-linux-gnu/bin/ld: cannot find -lfmtd: No such file or directory
collect2: error: ld returned 1 exit status
```

Related issues: #5970 , #966

### Steps to reproduce

1. Visit CE
2. Select an EDG-GNU compiler 
3. Add libfmt 10.2.1 to the libraries
4. Add an executor for the compiler
5. Build anything

### Expected behavior

It should be able to link+run when using libfmt.

### Reproduction link

https://gcc.godbolt.org/z/jEj46YW5h

### Screenshots

Not applicable

**Comment by partouf** [comment]
No, the EDG is not recognized as a working compiler by cmake https://conan.compiler-explorer.com/getlogging/20240331133436
Suggestions how to fix are welcome

**Comment by partouf** [comment]
The compiler seems to not support -Wl,rpath https://github.com/compiler-explorer/compiler-explorer/blob/f710f316cd41a2b6cabfd4aac676e9fb4df53f4d/lib/compilers/edg.ts#L37
We can probably work around that

**Comment by partouf** [comment]
I have fixed the rpath issue, but a new issue has popped up that I can't fix.
https://conan.compiler-explorer.com/getlogging/20240412030928
